### DISCUSSION TITLE
how is CSS handled?

### DISCUSSION THREAD

**Discussion by merodiro** [OpenSourceRaidGuild/website#79] [General]
as @nobrayner suggested we need to decide how are we going to handle CSS so we can document it.

things we need to decide:
- are we going to continue with emotion or use tailwind? and are we going to use emotion but have tailwind design system values?
- if we are going to build our design system how are going to handle spacing `em` or `rem` or both?
- how will the spacing scale look like?

feel free to add anything I missed

**Comment by datner** [comment]
My vote is on tailwind

**Reply by datner** [reply]
I want to use something that as documented well, and is well known so many people have an easy time working with it, also something very opinionated so developers have limited input on convention, reducing overhead and discussions

**Reply by JacobMGEvans** [reply]
+1 Tailwind

**Reply by datner** [reply]
@JacobMGEvans don't +1 in the comments, theres a literal voting mechanism in this feature, +1 my comment 😆

**Comment by AhmedEldessouki** [comment]
My vote goes for emotion. in my opinion and according to what Kent said, ready to use systems might come with limits. in the other hand if we continued with emotion, it will be easy to refactor or improve according to our needs.

I don't have any experience with em but I dont mind using it or both.

I leave the third for i will go with the majority

**Reply by datner** [reply]
an important distinction is to be made, tailwind is not a ready-to-use system. These are attributed to material-ui, ant-design, react-bootstrap, blueprint-ui, semantic-ui, and so on and so forth. tailwind is a css utils library, and we can use it alongside regular css. I disagree with Kents opinion about these, but that is another discussion altogether

**Reply by AhmedEldessouki** [reply]
I agree it is another discussion and i didnt place tailwind with material-ui and the others. its more flexible than them but in my opinion emotion is more flexible than tailwind 🙂 sorry if wasnt clear in my first comment =)

**Reply by merodiro** [reply]
Kent prefers The CSS Prop syntax for emotion which is so close to tailwind
he also mentioned that he doesn't like the styled syntax or CSS modules because that forces him to choose a name for everything even when naming doesn't make sense

also I think Kent mentioned sometime before that if he were to start building the bookshelf app again he would have just picked tailwind

**Reply by AhmedEldessouki** [reply]
i do remember the second one. the third i must have missed it. thu im still on the emotion side 😄 
thanks for the clarification 👍🏼 🙏🏼

**Reply by JacobMGEvans** [reply]
We definitely are not using Component Libraries like Material UI lol

**Reply by datner** [reply]
>doesn't like the styled syntax or CSS modules because that forces him to choose a name for everything even when naming doesn't make sense

I agree with this so much. It's the sole reason why I dropped styled-component and didn't ever recommend it to any of my clients

**Reply by AhmedEldessouki** [reply]
> 
> 
> We definitely are not using Component Libraries like Material UI lol

I never said we do 😄 i cleared that later on 😆

**Reply by AhmedEldessouki** [reply]
> 
> 
> > doesn't like the styled syntax or CSS modules because that forces him to choose a name for everything even when naming doesn't make sense
> 
> I agree with this so much. It's the sole reason why I dropped styled-component and didn't ever recommend it to any of my clients

+1

**Comment by merodiro** [comment]
My vote is on tailwind mainly because it comes with good defaults so we don't need to have a discussion on `em` vs `rem` or picking names for the variables that are meaningful to everyone. and also because it's already documented so we don't have to worry about creating documentation for our custom design system.

**Reply by JacobMGEvans** [reply]
+1 Tailwind as well
